deltachat-ffi/deltachat.h

```diff
@@ -5533,6 +5598,15 @@ void dc_event_unref(dc_event_t* event);
 #define DC_EVENT_MSGS_CHANGED             2000
 
 
+/**
+ * Message reactions changed.
+ *
+ * @param data1 (int) msg_id ID of the message for which reactions were changed.
```

Comment:
the combination of `chat_id` and `msg_id`, as for similar events,  seems more practical. eg. on ios/android, the `chat_id` is used to quickly check if an event is of interest at all (otherwise a db call would be needed).

the `contact_id` is probably only useful in the reaction details, where one would probably re-render everythings anyway (as also the tabs showing the count need an update).
same for the summaries in the bubbles, having the `contact_id` in quick access  is not really helpful to update sth. as `👍12 ❤️4` - one have to iterate over all reactions anyway.

and, if we later encounter we need `contact_id` as well, we can introduce a data3 :)

Reply:
> chat_id is not known when the event is emitted, so a database call would be needed anyway if you want to know the chat

if an additional database call is needed, i am also be against the proposed change.

however, having a look at the code, an additional database call seems not needed:

when calling `set_msg_reaction()` the `chat_id` is known as well as when calling `set_msg_id_reaction()` - it just needs to be passed to these methods.

> I expected that UI can check whether given msg_id is in the chatlist or not without the need to know chat_id, isn't it the case for iOS and Android?

well, in practice, eg. on android filtering is based by [the event parameters](https://github.com/deltachat/deltachat-android/blob/master/src/org/thoughtcrime/securesms/ConversationFragment.java#L955), the adapter contains the message-ids-array, but exposes only methods for getting count and msg_id by an index.

of course, that could be changed, but that is more complex than changing core and would make adaption harder, the REACTIONS event would need to be handled differently etc. and this is only android.

also, even if would do all the above and use an existing array, that may contain some thousands messages, iterating that over and over is still a waste of performance - esp. if we skip the `chat_id` also for future event additions or would have done that in the past (and that could not easily debounced as this is about the decision if the event "is ours", optimizing would require additional data types, again complexity ...)

tl;dr: to keep adaption straight-forward, simple and performant, i recommend passing `chat_id`+`msg_id` :)

> `contact_id` is not so useful maybe, it's easier to reload all message reactions anyway, but may be useful for bots who want to know who reacted to their message.

well, that is a theoretical aspect of one possible future  - whereas the above is very real. and, as said above,  we can  add `contact_id` nevertheless at some point (there are also other events where more details may be useful, however, in general, getting details about changes is not up to the event cmp. CONNECTIVITY_CHANGED)